tensorflow/python/keras/metrics.py

```diff
@@ -877,7 +877,7 @@ def update_state(self, y_true, y_pred, sample_weight=None):
     The shapes of `y_true` and `y_pred` are different.
 
     Args:
-      y_true: Ground truth label values. shape = `[batch_size, 1]`.
+      y_true: Ground truth label values. shape = `[batch_size, d1, .. dN-1]` or shape = `[batch_size, d1, .. dN-1, 1]`.
```

Comment:
From the existing code, 
https://github.com/tensorflow/tensorflow/blob/9410d39d0a9f850c4fb615a1d8b607f1b75d80c3/tensorflow/python/keras/metrics.py#L634-L644
the dimension start from `d0` and ends at `dN`, counting `N+1` dimensions in total. Is this the correct way of annotation dimensions?

If it is `N+1` dimensions by design, to make the dimension annotation consistent with the above code in `metric.py`, here it should be `[batch_size, d0, .. dN-1]` or `[batch_size, d0, .. dN-1, 1]`. 

If it is `N` dimensions, here it should be `[batch_size, d0, .. dN-2]` or `[batch_size, d0, .. dN-2, 1]`, and other occurrences should be corrected as well.

Reply:
All right then, in that case, the docstring here should be
```suggestion
      y_true: Ground truth label values. shape = `[batch_size, d0, .. dN-1]` or shape = `[batch_size, d0, .. dN-1, 1]`.
```